.github/workflows/bcny-build-nuget-package.yml

```diff
@@ -0,0 +1,142 @@
+name: "bcny-build-nuget-package"
+on:
+  workflow_call:
+    inputs:
+      arch:
+        description: "Name of the architecture amd64 or arm64"
+        required: true
+        type: string
+      platform:
+        description: "MSVC understandable platform name x64 or ARM64"
+        required: true
+        type: string
+
+jobs:
+  build_and_package:
+    name: "Build and package libsodium"
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: compnerd/gha-setup-vsdevenv@main
+        with:
+          host_arch: amd64
+          components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
+          arch: ${{ inputs.arch }}
+
+      - uses: actions/setup-python@v4
+        id: python
+        with:
+          python-version: 3.9
+          architecture: 'x64'
+
+      - name: Build StaticRelease
+        run:
+          msbuild .\builds\msvc\vs2022\libsodium.sln /p:Configuration=StaticRelease /p:Platform=${{ inputs.platform }}
+
+      - name: Build DynRelease
+        run:
+          msbuild .\builds\msvc\vs2022\libsodium.sln /p:Configuration=DynRelease /p:Platform=${{ inputs.platform }}
+
+      - name: Build LtcgRelease
+        run:
+          msbuild .\builds\msvc\vs2022\libsodium.sln /p:Configuration=LtcgRelease /p:Platform=${{ inputs.platform }}
```

Comment:
Is this a dll or static version?  I also wonder if this actually works with non-link linkers.

Reply:
I believe it's static. Not sure about non-link linkers as I don't know what they are :)

Reply:
The only ones that matter are non-link linkers - we use lld for linking Swift code.

Reply:
I really would love to know what `non-link linkers` are, I tried googling that, but there is no combination of this words on the internet according to Google, so I'm kinda confused.

<img width="245" alt="image" src="https://github.com/thebrowsercompany/libsodium/assets/297223/409bf7e7-9a17-4db6-9e50-d172ce675867">

Reply:
I was referring to lld (as I mentioned in https://github.com/thebrowsercompany/libsodium/pull/2#discussion_r1462019546), but BFD is another. I was being lazy and didn't quote it properly, I meant non-`link` linkers, aka, linkers other than `link.exe` which is Microsoft's linker.

Reply:
I see, I guess it doesn't matter for the `ltcg` configuration as we are not shipping it, but I was successful in linking that `static` as well as `dynamic` with `lld`.

Reply:
Right, the issue with the LTCG is that it depends on the compiler and linker working in consort, so built with LTCG may not be possible to use with clang/lld. While using LTCG would be amazing, I don't think that we are in a position to do that currently. If were packaging it, we should ensure that it was usable in the build as it would trip up people otherwise.

Reply:
Makes sense! Thank you for the explanation!
As you requested I limited the nuget package to `static` configuration only, so this is non-issue for now.